core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -829,7 +828,7 @@ private[spark] class TaskSchedulerImpl(
    * Get a snapshot of the currently blacklisted nodes for the entire application.  This is
    * thread-safe -- it can be called without a lock on the TaskScheduler.
    */
-  def nodeBlacklist(): scala.collection.immutable.Set[String] = {
+  def nodeBlacklist(): Set[String] = {
     blacklistTrackerOpt.map(_.nodeBlacklist()).getOrElse(scala.collection.immutable.Set())
```

Comment:
You can change this line too. I agree, I don't see the reason to use a mutable-or-immutable Set here. It's also used in getExecutorsAliveOnHost but don't see why.

Reply:
There's also a lot of import of `scala.collection.Map` in other classes, but that looks more legit: pattern matching, perhaps Java interop.

Oh, I wonder if this will cause the signature of getExecutorsAliveOnHost to change... may need a MiMa exclusion.

Reply:
This is an internal class, so there is no mima issue.